message/retrylater.go

```diff
@@ -21,10 +20,13 @@ type retryLaterHandler struct {
 }
 
 func (r *retryLaterHandler) Do(ctx context.Context, orig Handler, message *servicebus.Message) Handler {
-	select {
-	case <-ctx.Done():
-		return Error(fmt.Errorf("aborting retrylater for message. abandoning meddage %s: %w", message.ID, ctx.Err()))
-	case <-time.After(r.retryAfter):
-		return orig
-	}
+	go func() {
```

Comment:
is the new thread there to make it return faster? but then we may lose the ctx timeout error.

Reply:
yes the go function means we don't block other items but it does make it harder to timeout with conext. Not sure we shoudl even check context.

Reply:
Correct me if I'm wrong, but I don't think returning early here is necessarily useful, and possibly not actually a good idea.

If I understand it correctly, the message will still be locked on the topic until the timeout and the message is abandoned, so will not be able to pick up a new message until the time elapses anyways.

Plus, in my digging I spotted what seem to be a setting in azure-service-bus-go where the handleMessage function within side of azure-service-bus-go will handle the Disposition. I don't completely understand how the Disposition code works, but it makes me think that it could be possible that the thread is waiting here to abandon, but by returning the listener actually disposes of the msg from the topic. I don't believe this is currently being used by go-shuttle or exposed as a setting, but by returning here instead of letting the Abandon play out it seems like it could result in unexpected behavior down the line if more changes are made to the repo.

Reply:
From the most recent test I ran I think my assumption here about how the locking was incorrect. I detailed why I think my assumption was wrong in the chat.